mulighetsrommet-api/src/main/kotlin/no/nav/mulighetsrommet/api/services/NavAnsattService.kt

```diff
@@ -29,6 +32,35 @@ class NavAnsattService(
         return navAnsattRepository.getAll(roller = filter.roller)
     }
 
+    suspend fun addUserToKontaktpersoner(navIdent: NavIdent, tx: Session) {
+        val kontaktPersonGruppeId = roles.find { it.rolle == NavAnsattRolle.KONTAKTPERSON }?.adGruppeId
+        requireNotNull(kontaktPersonGruppeId)
+
+        val ansatt = navAnsattRepository.getByNavIdent(navIdent)
+            ?: microsoftGraphService.getNavAnsattByNavIdent(navIdent, AccessType.M2M)?.let {
+                NavAnsattDto.fromAzureAdNavAnsatt(it, roller = emptySet())
+            }
+        requireNotNull(ansatt) {
+            "Fant ikke ansatt med navIdent=$navIdent i AzureAd"
+        }
```

Comment:
Kunne kanskje trukket ut dette til en egen funksjon ala `getOrSynchronizeNavAnsatt` definert over? For lesbarhetens skyld? :smile:

Reply:
Heller litt mot å ikke gjøre det. Fordi hvis add kallet feiler så skal man ikke lagre (kunne fortsatt gjort det i transaskjon da), og da må vi lagre først og så lagre gruppen rett etterpå.

Så kanskje vente litt, se om det vil bli brukt andre steder?